### PR TITLE
Add router shutdown

### PR DESCRIPTION
**Pull request by gterzian**
Adding a shutdown operation to the router, see for context https://github.com/servo/servo/pull/25685#pullrequestreview-364604068

### CODE REVIEW

**Comment by gterzian** [comment]
@asajeffrey r?

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ipc-channel"
-version = "0.14.0"
+version = "0.14.1"
```
Comment (CYBAI) [inline Cargo.toml]: The `RouterMsg` enum is added one more variant. I think it might be good to upgrade the minor version instead 👀?

Comment (gterzian) [reply Cargo.toml]: The enum is private, so I don't think it changes the api?

Comment (CYBAI) [reply Cargo.toml]: Ahh, I see; didn't notice that. sorry, then ignore my comment >_<

**Comment by gterzian** [comment]
Oops, looks like a `cargo fmt` went through on the commit as well...

File: src/router.rs [hunk 2]
```diff
@@ -53,19 +54,52 @@ impl RouterProxy {
                 msg_sender: msg_sender,
                 wakeup_sender: wakeup_sender,
             }),
+            shutdown: Mutex::new(false),
```
Comment (asajeffrey) [inline src/router.rs]: Any time I see two locks, my deadlock alarms go off. Any chance you could make this an `AtomicBool` instead?

Comment (gterzian) [reply src/router.rs]: Good point, and what I've done is put the boolean into the other locked data structure, what do you think(I find something like `compare_and_swap` really quite hard to work with)?

**Comment by gterzian** [comment]
(I've also tried to make the test a bit clear, by only calling `shutdown` a second time at the end and adding a comment as to why).

**Comment by gterzian** [comment]
@asajeffrey r? I've addressed your comment, although in a slightly different way from your initial suggestion. Please let me know what you think...

**Comment by asajeffrey** [comment]
LGTM. @bors-servo r+

**Pull request merged**